### DISCUSSION TITLE
How to generate full static HTML site? No next.js javascript in it. just pure HTML thing

### DISCUSSION THREAD

**Discussion by liesislukas** [vercel/next.js#15996] [Help]
I have general page [...slug].js with idea to grab slug from router and then build the content. I just get the content from backend by slug. Slug may be `cat-1/cat-1.1/article` or just `article`. 

so inside `[...slug].js` i have `getStaticPaths` which fetches the article data and renders it. 

- First of all it was working yet it was not HTML, it was HTML+nextJs data tag and JS was required to render the page. 
- Then i tried to use `exportPathMap` inside `next.config.js` and i started to get the article rendered as it should yet then menu was not rendering after navigating into one of the articles and refreshing the page. Menu was still rendering with JS requirement.

![image](https://user-images.githubusercontent.com/2733862/89705314-03638700-d965-11ea-9332-5f01fbb5e41a.png)

in some git issues, I read that exportPathMap is depreciated. Yet not sure how to render the static HTML.

I tried adding this code to `[...slug].js` but then i again had partial render. All dynamic content coming from backend was not rendered into HTML.

```js
export const config = {
  unstable_runtimeJS: false,
};
```

also tried `getStaticPaths` inside `[...slug].js` with `fallback: false`

Is there any guide where it would say how to render a full static HTML page w/o any JS in the deployed production site?
I guess i used all good things for the job but in some wrong combination

**Comment by jnv** [comment]
I don't have experience with this use case, but from the little I know it seems that `unstable_runtimeJS` (from #11949) should be your best bet for now. But it's marked as unstable and undocumented for a reason.

You are mentioning issues with incomplete rendering of particular components when the runtime scripts are not present. Perhaps if you reproduced the issue in some minimal example others may be able to help? You can, for example, swap CMS for static values or static JSON files.

**Reply by liesislukas** [reply]
i tried it with no luck. some parts were rendering some not. what i do now is i build nextjs code with a shell script and then just not useing dynamic routs or prefetch of any props anywhere. sh scripts put json straight in js files

**Comment by jamesmosier** [comment]
[next-client-script](https://github.com/amannn/next-client-script) is a project that is working towards this goal.
